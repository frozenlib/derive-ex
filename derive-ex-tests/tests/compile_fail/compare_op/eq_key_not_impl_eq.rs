use derive_ex::derive_ex;

#[derive_ex(Eq, PartialEq)]
struct X {
    #[eq(key = $.0)]
    x: (f64, f64),
}

fn main() {}
