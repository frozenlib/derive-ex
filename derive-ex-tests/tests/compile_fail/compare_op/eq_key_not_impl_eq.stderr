error[E0277]: the trait bound `f64: std::cmp::Eq` is not satisfied
 --> tests/compile_fail/compare_op/eq_key_not_impl_eq.rs:5:16
  |
5 |     #[eq(key = $.0)]
  |                ^ the trait `std::cmp::Eq` is not implemented for `f64`
  |
  = help: the following other types implement trait `std::cmp::Eq`:
            isize
            i8
            i16
            i32
            i64
            i128
            usize
            u8
          and $N others
note: required by a bound in `_eq`
 --> tests/compile_fail/compare_op/eq_key_not_impl_eq.rs:5:16
  |
5 |     #[eq(key = $.0)]
  |                ^ required by this bound in `_eq`
